client/app/vps/dashboard/vps-dashboard.controller.js

```diff
@@ -308,6 +304,24 @@ class VpsDashboardCtrl {
         this.ControllerHelper.navigation.getConstant("changeOwner").then(url => { this.actions.changeOwner.href = url; });
     }
 
+    getRegionsGroup (regions) {
+        this.regionsGroup = [];
+        if (regions) {
+            this.detailedRegions = !_.isArray(regions) ?
+                [this.RegionService.getRegion(regions)] :
+                _.map(regions, region => this.RegionService.getRegion(region));
+        }
+        this.regionsGroup = _.groupBy(this.detailedRegions, "country");
+    }
+
+    hasMultipleRegions () {
+        return _(this.detailedRegions).isArray() && this.detailedRegions.length > 1;
+    }
+
+    getActionStateParamString (params) {
+        if (!params) { return ""; }
+        return `(${JSON.stringify(params)})`;
+    }
```

Comment:
Considering this task is supposed to be refactoring, is it okay to add the Region feature in the same PR?
Just wonderin'

Reply:
@euhmeuh  The region feature is already there in <tile> component. I just moved it to controller so refactoring will hold good for this.